tom-swifty/SwiftyOptions.cs

```diff
@@ -261,38 +264,62 @@ public void CheckForOptionErrors (ErrorHandling errors)
+										path = Path.Combine (SwiftGluePath, "mac/XamGlue.framework");
+									} else if (targetOS.StartsWith ("ios", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "iphone/XamGlue.framework");
+									} else if (targetOS.StartsWith ("tvos", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "appletv/XamGlue.framework");
+									} else if (targetOS.StartsWith ("watchos", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "watch/XamGlue.framework");
+									}
+								} else {
+									// TJ
+									// this is where it XamGlue.framework appears for me
+									// not sure if this is library specific
+									if (targetOS.StartsWith ("macos", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "mac/FinalProduct/XamGlue.framework");
+									} else if (targetOS.StartsWith ("ios", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "iphone/FinalProduct/XamGlue.framework");
+									} else if (targetOS.StartsWith ("tvos", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "appletv/FinalProduct/XamGlue.framework");
+									} else if (targetOS.StartsWith ("watchos", StringComparison.Ordinal)) {
+										path = Path.Combine (SwiftGluePath, "watch/FinalProduct/XamGlue.framework");
+									}
 								}
 								if (path != null) {
 									ModulePaths.Add (path);
 									DylibPaths.Add (path);
 								}
 							}
-							if (targetOS.StartsWith ("macos", StringComparison.Ordinal)) {
-								SwiftLibPath = Path.Combine (SwiftLibPath, "macosx");
-							} else if (targetOS.StartsWith ("ios", StringComparison.Ordinal)) {
-								SwiftLibPath = Path.Combine (SwiftLibPath, "iphoneos");
-							} else if (targetOS.StartsWith ("tvos", StringComparison.Ordinal)) {
-								SwiftLibPath = Path.Combine (SwiftLibPath, "appletvos");
-							} else if (targetOS.StartsWith ("watchos", StringComparison.Ordinal)) {
-								SwiftLibPath = Path.Combine (SwiftLibPath, "watchos");
+							// TJ
+							// this group of conditionals look for different paths than what I see, unless the SwiftLibPath contains /lib/swift/
+							// for example:
+							// ../SwiftToolchain-v1-28e007252c2ec7217c7d75bd6f111b9c682153e3/build/Ninja-ReleaseAssert/swift-macosx-x86_64/lib/swift/iphoneos
+							if (!isLibrary) {
+								if (targetOS.StartsWith ("macos", StringComparison.Ordinal)) {
+									SwiftLibPath = Path.Combine (SwiftLibPath, "macosx");
+								} else if (targetOS.StartsWith ("ios", StringComparison.Ordinal)) {
+									SwiftLibPath = Path.Combine (SwiftLibPath, "iphoneos");
+								} else if (targetOS.StartsWith ("tvos", StringComparison.Ordinal)) {
+									SwiftLibPath = Path.Combine (SwiftLibPath, "appletvos");
+								} else if (targetOS.StartsWith ("watchos", StringComparison.Ordinal)) {
+									SwiftLibPath = Path.Combine (SwiftLibPath, "watchos");
```

Comment:
AFAIK you can use a switch with pattern matching. https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching